internal/metrics/sheets/sheets.go

```diff
@@ -0,0 +1,194 @@
+	serviceRow, err := p.retrieveServiceRow(logger)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to retrieve metrics for the service")
+	}
+
+	var col string
+	var ok bool
+	switch alignReduceType {
+	case metrics.Align99Reduce99:
+		col, ok = serviceRow[colLatencyP99].(string)
+		break
+	case metrics.Align95Reduce95:
+		col, ok = serviceRow[colLatencyP95].(string)
+		break
+	case metrics.Align50Reduce50:
+		col, ok = serviceRow[colLatencyP50].(string)
+		break
+	}
+
+	if !ok {
+		return 0, errors.New("invalid latency value, cell must be a string")
+	}
+	value, err := strconv.ParseFloat(col, 64)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to parse latency value")
+	}
+	return value, nil
+}
+
+// ErrorRate returns the rate of 5xx errors for the resource matching the filter.
+func (p *Provider) ErrorRate(ctx context.Context, offset time.Duration) (float64, error) {
+	logger := util.LoggerFromContext(ctx)
+	logger.Debug("querying google sheet for request count")
+	serviceRow, err := p.retrieveServiceRow(logger)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to retrieve metrics for the service")
+	}
+
+	col, ok := serviceRow[colErrorRate].(string)
+	if !ok {
+		return 0, errors.New("invalid error rate value, cell must be a string")
+	}
+	value, err := strconv.ParseFloat(col, 64)
+	if err != nil {
+		return 0, errors.Wrap(err, "failed to parse error rate value")
+	}
+	return value, nil
+}
+
+// retrieveServiceRow returns the row that contains the information about the
+// service
+func (p *Provider) retrieveServiceRow(logger *logrus.Entry) ([]interface{}, error) {
+	values, err := p.retrieveValues(logger)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve values")
+	}
+
+	serviceRow, err := p.filterServiceRow(values)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to filter service row")
+	}
+	if serviceRow == nil {
+		return nil, errors.New("no service matched the query")
+	}
+	return serviceRow, nil
+}
+
+// retrieveValues get all the metrics values starting at row 2.
+func (p *Provider) retrieveValues(logger *logrus.Entry) ([][]interface{}, error) {
+	readRange := "A2:G"
+	if p.sheetName != "" {
+		readRange = p.sheetName + "!" + readRange
+	}
+	resp, err := p.client.Spreadsheets.Values.Get(p.sheetsID, readRange).Do()
+	if err != nil {
+		return nil, errors.Errorf("unable to retrieve data from sheet: %v", err)
+	}
+	logger.Debugf("queried %d rows", len(resp.Values))
+
+	return resp.Values, nil
+}
+
+// filterServiceRow returns the first row that matches the region and service
+// name.
+func (p *Provider) filterServiceRow(values [][]interface{}) ([]interface{}, error) {
+	for _, row := range values {
+		region, ok := row[colRegion].(string)
+		if !ok {
+			return nil, errors.New("invalid region value, cell must be a string ")
+		}
+		serviceName, ok := row[colServiceName].(string)
+		if !ok {
+			return nil, errors.New("invalid service name value, cell must be a string ")
```

Comment:
:) you forgot to do errorf again. 
we will never know what value caused the error.

x2

Reply:
Fixed! I also did it for all the metrics values